ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/metacard-quality/metacard-quality.tsx

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+import styled from '../../styles/styled-components'
+import LoadingCompanion from '../../container/loading-companion'
+
+type Props = {
+  metacardValidation: any
+  attributeValidation: any
+  loading: boolean
+}
+
+const Header = styled.h4`
+  text-align: center;
+`
+
+const Root = styled.div`
+  overflow: auto;
+
+  table {
+    width: 100%;
+    text-align: center;
+    margin-bottom: 40px;
+  }
+
+  th {
+    width: 33%;
+    text-align: center;
+  }
+
+  tr:nth-of-type(even) {
+    background: rgba(0, 0, 0, 0.1);
+  }
+
+  th,
+  td {
+    padding: 10px;
+  }
+
+  td + td {
+    border-left: 1px solid rgba(100, 100, 100, 0.3);
+  }
+
+  tbody {
+    border-top: 1px solid rgba(100, 100, 100, 0.3);
+  }
+`
+
+const render = (props: Props) => {
+  const { metacardValidation, attributeValidation, loading } = props
+  return (
```

Comment:
This is a very large render. I'd recommend pulling out what you can into named functions that return jsx.